app/scripts/controllers/pod.js

```diff
@@ -87,6 +87,67 @@ angular.module('openshiftConsole')
       }
     };
 
+    /**
+     *  Will set the newTerm's isVisible/isUsed to true, while hiding the previous
+     */
+    $scope.onTerminalSelectChange = function(newTerm) {
+      // Make all terminals invisible (Becuase we don't have a pointer to the terminal that is currently visible)
+      _.each($scope.containerTerminals, function(term) {
+        term.isVisible = false;
+      });
+
+      newTerm.isVisible = true;
+      newTerm.isUsed = true;
+    };
+
+    $scope.tabSelect = function() {
+      $scope.terminalTabWasSelected = true;
+    };
+
+    var makeTerminals = function(existingTerminals) {
+      existingTerminals = existingTerminals || [];
+
+      // Remake list with old list's isVisible/isUsed data
+      _.each($scope.pod.spec.containers, function(spec) {
+        var terminal = _.find(existingTerminals, function(term) {
+          return term.containerName === spec.name;
+        });
+
+        if (!terminal) {
+          existingTerminals.push({
```

Comment:
Perhaps do the `_.find` once, then reuse the object:
```javascript
var found = _.find($scope.containerTerminals, function(term) {
     return term.containerName === spec.name;
};
 existingTerminals.push({
   isVisible: _.get(found, 'isVisible'
   // etc.
});
```
or, perhhaps a `_.merge({}, {})` kind of thing would work here so you dont have to repeat the key names `isVisible, isUsed, etc.` but lemme look a bit more.